source/utilities.cc

```diff
@@ -938,12 +948,166 @@ namespace aspect
+            {
+              libdap::Connect *url = 0;
+              url = new libdap::Connect(filename);
+              libdap::BaseTypeFactory factory;
+              libdap::DataDDS dds(&factory);
+              libdap::DAS das;
+
+              url->request_data(dds, "");
+              url->request_das(das);
+
+
+              //Array to store the url data
+              libdap::Array *urlArray;
+
+              //Temporary vector that will hold the different arrays stored in urlArray
+              std::vector<std::string> tmp;
+              //Vector that will hold the arrays (columns) and the values within those arrays
+              std::vector<std::vector<std::string>> columns;
+
+
+              //Check dds values to make sure the arrays are of the same length and of type string
+              for (libdap::DDS::Vars_iter i = dds.var_begin(); i != dds.var_end(); i++)
+                {
+                  libdap::BaseType *btp = *i;
+                  if ((*i)->type() == libdap::dods_array_c)
+                    {
+                      urlArray = static_cast <libdap::Array *>(btp);
+                      if (urlArray->var() != NULL && urlArray->var()->type() == libdap::dods_str_c)
+                        {
+                          //The url Array contains a separate array for each column of data.
+                          // This will put each of these individual arrays into its own vector.
+                          urlArray->value(tmp);
+                          columns.push_back(tmp);
+
+                        }
+                      else
+                        {
+                          AssertThrow (false,
+                                       ExcMessage (std::string("Error when reading from url: ") + filename +
+                                                   " Maybe it was not of the correct type?"));
+                        }
+
+                    }
+                  else
+                    {
+                      AssertThrow (false,
+                                   ExcMessage (std::string("Error when reading from url: ") + filename +
+                                               " Maybe it was not of the correct type?"));
+                    }
+                }
+
+
+              //Add the POINTS data that is required and found at the top of the data file.
+              // The POINTS values are set as attributes inside a table.
+              // Loop through the Attribute table to locate the points values within
+              std::vector<std::string> points;
+              libdap::AttrTable *table;
+              for (libdap::AttrTable::Attr_iter i = das.var_begin(); i != das.var_end(); i++)
+                {
+                  table = das.get_table(i);
+                  if (table->get_attr("points") != "")
+                    points.push_back(table->get_attr("points"));
+                  else
+                    break;
```

Comment:
why break here?

Reply:
I think that may have been leftover code. Removed it.